packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -75,14 +75,18 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
 
     $stopPlugin(contextPath: string): PromiseLike<void> {
         this.activatedPlugins.forEach(plugin => {
-            if (plugin.stopFn) {
-                plugin.stopFn();
-            }
-
-            // dispose any objects
-            const pluginContext = plugin.pluginContext;
-            if (pluginContext) {
-                dispose(pluginContext.subscriptions);
+            try {
+                if (plugin.stopFn) {
+                    plugin.stopFn();
+                }
+
+                // dispose any objects
+                const pluginContext = plugin.pluginContext;
+                if (pluginContext) {
+                    dispose(pluginContext.subscriptions);
+                }
+            } catch (error) {
+                console.error(error);
```

Comment:
Is it coming from `stopFn`? when only it should be wrapped, we should clean subscriptions always

Reply:
I stripped out extension of java and java-debug

Reply:
vscode-java version is 0.44.0
ms-python.python is 2019.3.6215

Reply:
Python Extension: Executing next rule from system
Python Extension: autoSelectInterpreter, Class name = g, Arg 1: undefined, Return Value: undefined
 root ERROR [hosted-plugin: 95] TypeError: d.dispose is not a function
    at /opt/theia/packages/plugin-ext/lib/common/disposable-util.js:24:52
    at Array.forEach (<anonymous>)
    at Object.dispose (/opt/theia/packages/plugin-ext/lib/common/disposable-util.js:24:15)
    at /opt/theia/packages/plugin-ext/lib/plugin/plugin-manager.js:112:39
    at Map.forEach (<anonymous>)
    at PluginManagerExtImpl.$stopPlugin (/opt/theia/packages/plugin-ext/lib/plugin/plugin-manager.js:104:31)
    at RPCProtocolImpl.doInvokeHandler (/opt/theia/packages/plugin-ext/lib/api/rpc-protocol.js:207:23)
    at RPCProtocolImpl.invokeHandler (/opt/theia/packages/plugin-ext/lib/api/rpc-protocol.js:192:41)
    at RPCProtocolImpl.receiveRequest (/opt/theia/packages/plugin-ext/lib/api/rpc-protocol.js:154:45)
    at RPCProtocolImpl.receiveOneMessage (/opt/theia/packages/plugin-ext/lib/api/rpc-protocol.js:123:22)

Reply:
Mr kosyakov：

This exception will cause the stopFn function to throw an error and affect the shutdown of other lsp processes.

Reply:
So I think it is necessary to catch the exception